src/workspace/mod.rs

```diff
@@ -296,26 +296,255 @@ impl FolderConfigs {
 #[derive(Debug)]
 pub struct Workspace {
   root_dir: UrlRc,
-  /// The directory the user started the workspace discovery from.
-  start_dir: UrlRc,
   config_folders: IndexMap<UrlRc, FolderConfigs>,
-  vendor_dir: Option<PathBuf>,
 }
```

Comment:
This looks like a lot of changes, but it's just a re-org. I've separated out `Workspace` from `WorkspaceContext`. The workspace only has information about the deno.jsons, packages.jsons, and folders, while the `WorkspaceContext` can be used to get the config information.